### PR TITLE
Create prow jobs for Cluster-API-Provider-Cloudstack sig repo

### PR DESCRIPTION
**Pull request by Josh-Bonello**
As part of migrating [cluster-api-provider-cloudstack](https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack) repository to kubernetes-sigs GitHub organization, we are migrating the prow based CI tools to leverage [Kubernete's maintained prow](https://prow.k8s.io/).

The contents of this pull request establish an OWNERS file for prow jobs ongoing and land the presubmit jobs which function on privately controlled prow instances and with local testing via `pj-on-kind.sh` transformations.

/sig cluster-lifecycle 
/cc @fabriziopandini @rejoshed

### CODE REVIEW

**Comment by Josh-Bonello** [comment]
I have requested CLA from my Corporate sponsor, awaiting approval.

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (rejoshed) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:99]: I think these two lines should be omitted. Otherwise without running this myself, it looks good.

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:1]: Wrong boilerplate

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/OWNERS
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/OWNERS:5]: it seems maxdrib and rejoshed are not member of the kubernetes org

Comment (Josh-Bonello) [reply config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/OWNERS]: I have removed these individuals while they apply for membership to kubernetes, they are currently members of `kubernetes-sigs`.

**Comment by Josh-Bonello** [comment]
It occurs to me that we also need a place to the prow k8s secrets that would not be committed to code. What is the procedure to add and maintain these secrets?

**Comment by rohityadavcloud** [comment]
Requesting owners/approvers to help assist, review cc @fabriziopandini @cblecker @spiffxp @wojtek-t @chases2 @dims

**Review by rohityadavcloud — APPROVED** [review]

**Review by fabriziopandini — COMMENTED** [review]
I don't have context on cloudstack provider stuff but overall seems ok to me
/lgtm

**Comment by fabriziopandini** [comment]
Please squash commits

**Comment by rohityadavcloud** [comment]
cc @fabriziopandini

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:49]: q: by a quick comparison with CAPI pre submit jobs it is missing the target branch. is it intentional?

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:54]: Is this intentional (jobs are never running automatically)?

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:60]: Is this intentional (skip setting a status on GitHub)

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (fabriziopandini) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:61]: Is this required? if I'm not wrong 
```suggestion
presubmits:
  kubernetes-sigs/cluster-api-provider-cloudstack:
```
Is enough

**Review by fabriziopandini — COMMENTED** [review]
Asking some question about job configuration after comparison with CAPI jobs, but I'm not an expert and an opinion from someone that knows better test-infra internals could really help here

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (Josh-Bonello) [reply config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:54]: Yes. This job is to be used for debugging "project build" purposes rather than for every PR update.

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (Josh-Bonello) [reply config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:49]: Per [presumbit spec's](https://pkg.go.dev/k8s.io/test-infra/prow/config#Presubmit) [Brancher spec](https://pkg.go.dev/k8s.io/test-infra/prow/config#Brancher)
```// Only run against these branches. Default is all branches.
	Branches [][string](https://pkg.go.dev/builtin#string) `json:"branches,omitempty"```

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (Josh-Bonello) [reply config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:61]: Tested and verified, pushed an update to omit the `clone_uri` line.

**Comment by fabriziopandini** [comment]
Thanks for clarifications
/lgtm
/approve

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:3]: this name is pretty ambiguous as we have a lot of "image" building jobs that look nothing like this.

can we please name it something more specific?

File: config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/cluster-api-provider-cloudstack/capi-provider-cloudstack-presumbit.yaml:40]: FYI: this preset is not set to key off of anything, please don't add environment variables to all jobs like this (!)

presets are global to the job config.

**Pull request merged**
